Documentation/design-docs/COM-activation.md

```diff
@@ -0,0 +1,132 @@
+# COM Activation for .NET Core on Windows
+
+## Purpose
+
+In order to more fully support the vast number of existing .NET Framework users in their transition to .NET Core, support of the COM activation scenario in .NET Core is required. Without this support it is not possible for many .NET Framework consumers to even consider transitioning to .NET Core. The intent of this document is to describe aspects of COM activation for a .NET class written for .NET Core. This support includes but is not limited to activation scenarios such as the [`CoCreateInstance()`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-cocreateinstance)API in C/C++ or from within a [Windows Script Host](https://docs.microsoft.com/en-us/windows/desktop/com/using-com-objects-in-windows-script-host) instance.
+
+COM activation in this document is currently limited to in-proc scenarios. Scenarios involving out-of-proc COM activation are deferred.
+
+### Requirements
+
+* Discover all installed versions of .NET Core.
+* Load the appropriate version of .NET Core for the class if a .NET Core instance is not running, or validate the currently existing .NET Core instance can satisfy the class requirement.
+* Instantiate an instance of the class and return a pointer to an [`IUnknown`](https://docs.microsoft.com/en-us/windows/desktop/api/unknwn/nn-unknwn-iunknown) that represents the newly constructed class.
+* Support the discrimination of concurrently loaded CLR versions.
+
+### Environment Matrix
+
+The following list represents an exhaustive activation matrix.
+
+| Server | Client | Current Support |
+| --- | --- | :---: |
+| COM* | .NET Core | Yes |
+| .NET Core | COM* | No |
+| .NET Core | .NET Core | No |
```

Comment:
how does these managed(.NET Core/Framework) to manage(.NET Core/Framework) need activation support?

Reply:
It hasn't been defined at the moment. It is same issue with .NET Core and .NET Core - what is the mechanism for when a .NET Core application activates a class that was defined by .NET Framework?

The point of that matrix is to be complete, we may not support certain scenarios, but I want to call them out.